### PR TITLE
Extract common tests for log and span filtering

### PR DESCRIPTION
**Pull request by zeitlinger**
follow up for #1934

### CODE REVIEW

**Comment by zeitlinger** [comment]
@tigrannajaryan this is now rebased and can be merged

File: internal/processor/filtermatcher/filtermatcher_test.go [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			output, err := NewMatcher(&tc.property)
+			assert.Zero(t, output)
+			assert.EqualError(t, err, tc.errorString)
+		})
+	}
+}
+
+func TestSpan_Matching_False(t *testing.T) {
```
Comment (tigrannajaryan) [inline internal/processor/filtermatcher/filtermatcher_test.go:118]:
```suggestion
func Test_Matching_False(t *testing.T) {
```
Please also remove the "Span" from the names of the tests that are generic filter tests and not span-specific.

**Comment by tigrannajaryan** [comment]
@zeitlinger lgtm, except the minor comment about usage of "Span" in the extracted tests. Please fix and can merge.

File: internal/processor/filtermatcher/filtermatcher_test.go (hunk 1 above)
Comment (zeitlinger) [reply internal/processor/filtermatcher/filtermatcher_test.go]: > @zeitlinger lgtm, except the minor comment about usage of "Span" in the extracted tests. Please fix and can merge.

done

**Review by tigrannajaryan — APPROVED** [review]

**Pull request merged**